test/unit-test/FreeRTOS_IPv6_Utils/FreeRTOS_IPv6_Utils_utest.c

```diff
@@ -0,0 +1,367 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+
+/* Include Unity header */
+#include "unity.h"
+
+/* Include standard libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "mock_task.h"
+#include "mock_list.h"
+
+/* This must come after list.h is included (in this case, indirectly
+ * by mock_list.h). */
+#include "mock_queue.h"
+#include "mock_event_groups.h"
+
+#include "mock_FreeRTOS_IP.h"
+
+#include "FreeRTOS_IPv6_Utils.h"
+#include "FreeRTOSIPConfig.h"
+
+#ifndef ipconfigTCPv6_MSS
+    #define ipconfigTCPv6_MSS    ( ipconfigNETWORK_MTU - ( ipSIZE_OF_IPv6_HEADER + ipSIZE_OF_TCP_HEADER ) )
+#endif
+
+/* Setting IPv6 address as "fe80::7009" */
+const IPv6_Address_t xDefaultIPAddress =
+{
+    0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x70, 0x09
+};
+
+#define ipSIZE_OF_IPv6_PAYLOAD_LEN    20
+
+#define ipICMP_Incorrect_Msg_IPv6     0
+
+/* Buffer length check of ICMPv6 packet failed. */
+#define ipFail_INVALID_LENGTH         10
+
+/*
+ * ===================================================
+ *        Test for vSetMultiCastIPv6MacAddress
+ * ===================================================
+ */
+
+
+/**
+ * @brief This function verify correctly setting
+ *        a MAC address.
+ */
+void test_vSetMultiCastIPv6MacAddress( void )
+{
+    IPv6_Address_t xIPAddress;
+    MACAddress_t xMACAddress;
+
+    /* Setting IPv6 address as "fe80::7009" */
+    memcpy( &xIPAddress, &xDefaultIPAddress, sizeof( IPv6_Address_t ) );
+    vSetMultiCastIPv6MacAddress( &xIPAddress, &xMACAddress );
+
+    TEST_ASSERT_EQUAL( ( uint8_t ) 0x33U, xMACAddress.ucBytes[ 0 ] );
+    TEST_ASSERT_EQUAL( ( uint8_t ) 0x33U, xMACAddress.ucBytes[ 1 ] );
+    TEST_ASSERT_EQUAL( xIPAddress.ucBytes[ 12 ], xMACAddress.ucBytes[ 2 ] );
+    TEST_ASSERT_EQUAL( xIPAddress.ucBytes[ 13 ], xMACAddress.ucBytes[ 3 ] );
+    TEST_ASSERT_EQUAL( xIPAddress.ucBytes[ 14 ], xMACAddress.ucBytes[ 4 ] );
+    TEST_ASSERT_EQUAL( xIPAddress.ucBytes[ 15 ], xMACAddress.ucBytes[ 5 ] );
+}
+
+/*
+ * ===================================================
+ *          Test for prvChecksumIPv6Checks
+ * ===================================================
+ */
+
+
+/**
+ * @brief This function validate handling of buffer length
+ *        less than the minimum packet size.
+ */
+void test_prvChecksumIPv6Checks_InvalidLength( void )
```

Comment:
prvChecksumICMPv6Checks returns 0 in case of Success, non-zero in case of error
So pdFALSE/pdTRUE is not the return value of prvChecksumICMPv6Checks

Reply:
Created macro and added